custom/icds_reports/static/js/directives/info-message/info-message.directive.js

```diff
@@ -4,21 +4,33 @@ var url = hqImport('hqwebapp/js/initial_page_data').reverse;
 
 function InfoMessageController($location) {
     var vm = this;
+    vm.previousToPreviousMonth = null;
+    vm.previousMonth = null;
     vm.currentMonth = null;
-    vm.lastDayOfPreviousMonth = null;
+    vm.nextMonth = null;
+    vm.year = null;
 
     vm.showInfoMessage = function () {
+        var start = parseInt(vm.start || '1');
+        var end = parseInt(vm.end || '2');
+        vm.type = vm.type || 'default';
+
         var selectedMonth = parseInt($location.search()['month']) || new Date().getMonth() + 1;
         var selectedYear = parseInt($location.search()['year']) || new Date().getFullYear();
         var currentMonth = new Date().getMonth() + 1;
         var currentYear = new Date().getFullYear();
-        if (!$location.path().startsWith("/fact_sheets") && !$location.path().startsWith("/download") &&
-            selectedMonth === currentMonth && selectedYear === currentYear &&
-            (new Date().getDate() === 1 || new Date().getDate() === 2)) {
-            vm.lastDayOfPreviousMonth = moment().set('date', 1).subtract(1, 'days').format('Do MMMM, YYYY');
+        var currentDate = new Date().getDate();
+
+        if (selectedMonth === currentMonth && selectedYear === currentYear && (currentDate >= start && currentDate <= end)) {
```

Comment:
this complex conditional could be well-suited to extraction (ether as a variable or function). e.g.

```
// not sure if this is the right name
var dateIsInCurrentMonth = (selectedMonth === currentMonth && selectedYear === currentYear && (currentDate >= start && currentDate <= end))
if (dateIsInCurrentMonth) {
...
}
```

Reply:
@czue ,
You're right, I'll separate the condition to a var.